### PR TITLE
Reimplement Field64 with u64 as internal storage

### LINKED ISSUE #1012: Narrower internal representation of Field64
Both Field64 and Field128 are internally represented using a u128. Multiplication for both is implemented by first splitting the internal representations of each multiplicand into two 64-bit limbs, cross-multiplying them to get 128-bit results, and combining and reducing from there. In the Field64 case, we know the high limbs are zero, so we are performing multiple unnecessary multiplications.

This isn't a big deal for Prio3Count, since the circuit and gadget polynomial are so small. However, with "multiproof" Prio3, the performance of Field64 multiplication matters more, since larger circuits over Field64 become relevant.

Setting aside fiat-crypto (which needs more complicated improvements to be competitive) I think the way to address this would be to split off a 64-bit version of `FieldParameters`, rewrite the Montgomery arithmetic routines, and add another identifier parameter to the `make_field!` macro.

### PR DESCRIPTION
**Pull request by divergentdave**
This PR adds an alternate implementation of field arithmetic using `u64`s in place of `u128`s. I found that this approximately doubles the speed of Prio3 when using a large circuit over Field64. I refactored the existing tests of FieldParameters to go through a testing-only trait, so I could reuse them to test the new implementation. I also updated the Sage script that produces precomputed constants, and cleaned up some small things in the existing FieldParameters methods.

This closes #1012.

### CODE REVIEW

**Review by inahga — APPROVED** [review]
Nicely done.

**Review by tgeoghegan — APPROVED** [review]

**Comment by cjpatton** [comment]
Before merging I'd suggest getting review from @armfazh, since he wrote this code.

**Comment by divergentdave** [comment]
I didn't like how `rand_elem()` round-tripped its random input through `montgomery()` and `residue()` before constructing a BigInt for the reference implementation of operations. I rewrote that test to instead convert the random input directly into a BigInt, so it can also catch bugs even if `montgomery()` and `residue()` aren't 1-1 and onto over [0, p).

**Comment by divergentdave** [comment]
I added bias to the randomized test, inspired by [this talk](https://youtu.be/lkEH3V3PkS0?t=1996), and that shook out a false positive when trying to take the inverse of zero.

**Review by branlwyd — APPROVED** [review]

**Review by armfazh — APPROVED** [review]
it looks good, the math didn't change, the only change was in being explicit on the radix used.

File: documentation/field_parameters.sage
Comment (divergentdave) [reply documentation/field_parameters.sage:29]: Done, thanks!

**Pull request merged**
